DOCSP-40960-compare-queries-task (#157)
mongodb/docs-relational-migrator: source/code-generation/query-converter/test-converted-queries/compare-converted-queries.txt

   :class: singlecol

You can use the :guilabel:`Test Queries` pane to compare source 
and destination queries, views, and stored procedure results 
inside of Relational Migrator. The :guilabel:`Test Queries` can 
help you verify the accuracy of converted code and
show the source and destination data after you run a sync job.

Before you Begin
----------------

- You must sign in to your Atlas account in Relational Migrator. 
  For details, see :ref:`rm-atlas-log-in`.

- For details on how to turn on Query Converter, see 
  :ref:`rm-enable-query-converter`.

- To use the :guilabel:`Test Queries` pane, Relational
  Migrator must have access to the Query Runner. For details
  on how to setup Query Runner, see :ref:`rm-install-query-runner`.

Steps
-----

.. procedure::
   :style: normal

   .. step:: Select a query on the :guilabel:`Query Converter` pane

      a. From the :guilabel:`Code Generation` tab, click the 
         :guilabel:`Query Converter` pane.

         .. note::

           If you are not already logged into your Atlas account or have
           an expired session, you must login to proceed.

      #. On the left-hand :guilabel:`Queries` pane, select a query, view,

>          or stored procedure.

Reviewer: Maybe add a step to convert the query to your chosen language, if it hasn't been converted already.

Author: Done.

   .. step:: Open the :guilabel:`Test Queries` pane

      a. From the :guilabel:`Code Generation` tab, click the 
         :guilabel:`Query Converter` pane.

      #. Click the :icon-lg:`ChevronUp` on the :guilabel:`Test Queries`
         pane at the bottom-right of the screen.

>    .. step:: (Optional) Specify parameters

Reviewer: It would be nice to have a clearer description of the 3 sections of the test bench: Console, Results and Messages. I'm not sure where this should sit relative to this procedure.

This step is about using the Console. The console is used for running all converted queries, and is also used for running relational stored procedures (but not queries or views). Specifying parameters is one of the things you need to do in the console window.

         LANGUAGE PLPQSQL

         AS $PROCEDURE$
         BEGIN 
            UPDATE ORDERS SET STATUS = 'CANCELLED' WHERE CUSTOMER_ID = CUST_ID;
         END;$PROCEDURE$

      Specify the ``CUST_ID`` in the :guilabel:`Source Script` text field
      by replacing ``<value>`` with the customer ID:

      .. code-block:: sql
         :copyable: false
         :emphasize-lines: 7

         DO 
         $$
            DECLARE
               CUST_ID INTEGER;
            BEGIN
               --TODO: Set parameter values here
               CUST_ID := <value>

               CALL public.cancel_customer_orders(cust_id);
            END;
         $$  

   .. step:: Run the source query

      a. On the :guilabel:`Test Queries`
         pane, click the :guilabel:`Run Source Query` 
         button.

      #. Enter the connection details to your source database.

      #. Click :guilabel:`Run` on the 
         :guilabel:`Connection Details` form.

         The data from the source database populates in 
         :guilabel:`Results`. You can click the :guilabel:`Messages` 
         pill to see execution statistics such as ``Execution Time``

>          and ``Row Count``.

Reviewer: Messages also shows any errors, as well as the output from statements in the query that print out messages.

.. _rm-compare-converted-queries:

=========================
Compare Converted Queries
=========================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can use the :guilabel:`Test Queries` pane to compare source 
and destination queries, views, and stored procedure results 

> inside of Relational Migrator. The :guilabel:`Test Queries` can 

Reviewer 2: [vale suggestion]

```suggestion
in Relational Migrator. The :guilabel:`Test Queries` can 
```

help you verify the accuracy of converted code and
show the source and destination data after you run a sync job.

About this Task
---------------

The :guilabel:`Test Queries` pane is split into three separate 

> user interfaces. Click the pill buttons next to 

Reviewer 2: [Suggestion]
```suggestion
user interfaces: Console, Results, and Messages. Click the pill buttons next to 
```

       a source or destination query to view the ``Results``
       user interface.
   * - Messages
     - This user interface is used to show messages from each 
       database connection. You can see error messages, 
       execution stats, and any print statements.

Before you Begin
----------------

- You must sign in to your Atlas account in Relational Migrator. 
  For details, see :ref:`rm-atlas-log-in`.

- For details on how to turn on Query Converter, see 
  :ref:`rm-enable-query-converter`.

- To use the :guilabel:`Test Queries` pane, Relational
  Migrator must have access to the Query Runner. For details
  on how to setup Query Runner, see :ref:`rm-install-query-runner`.

Steps
-----

.. procedure::
   :style: normal

   .. step:: Select a query on the :guilabel:`Query Converter` pane

      a. From the :guilabel:`Code Generation` tab, click the 
         :guilabel:`Query Converter` pane.

         .. note::

           If you are not already logged into your Atlas account or have
           an expired session, you must login to proceed.

      #. On the left-hand :guilabel:`Queries` pane, select a query, view,
         or stored procedure.

>       #. If your query has not been converted, select a 

Reviewer 2: [Question]

Should/could this step be marked as optional?

         :guilabel:`Target Language` and click :guilabel:`Convert`
         :icon-lg:`Sparkle`.

   .. step:: Open the :guilabel:`Test Queries` pane

      a. From the :guilabel:`Code Generation` tab, click the 
         :guilabel:`Query Converter` pane.

>       #. Click the :icon-lg:`ChevronUp` button on the 

Reviewer 2: [Suggestion]

Maybe to help users orient themselves a little quicker "On the bottom-right of the screen, Click the :icon-lg:`ChevronUp` button on the .."

         :guilabel:`Test Queries` pane at the bottom-right of 
         the screen.

   .. step:: (Optional) Specify parameters

>       If your converting a stored procedure with parameters, specify the 

Reviewer 2: ```suggestion
      If you're converting a stored procedure with parameters, specify the 
```

      For example, for the following PostGreSQL stored procedure:

      .. code-block:: sql
         :copyable: false

         CREATE OR REPLACE PROCEDURE PUBLIC.CANCEL_CUSTOMER_ORDERS(IN CUST_ID INTEGER)
         LANGUAGE PLPQSQL

         AS $PROCEDURE$
         BEGIN 
            UPDATE ORDERS SET STATUS = 'CANCELLED' WHERE CUSTOMER_ID = CUST_ID;
         END;$PROCEDURE$

      Specify the ``CUST_ID`` in the :guilabel:`Source Script` text field
      by replacing ``<value>`` with the customer ID:

      .. code-block:: sql
         :copyable: false
         :emphasize-lines: 7

         DO 
         $$
            DECLARE
               CUST_ID INTEGER;
            BEGIN
               --TODO: Set parameter values here
               CUST_ID := <value>

               CALL public.cancel_customer_orders(cust_id);
            END;
         $$  

   .. step:: Run the source query

      a. On the :guilabel:`Test Queries`
         pane, click the :guilabel:`Run Source Query` 
         button.

      #. Enter the connection details to your source database.

      #. Click :guilabel:`Run` on the 
         :guilabel:`Connection Details` form.

>          The data from the source database populates in 

Reviewer 2: ```suggestion
         The data from the source database populates in the
```

         :guilabel:`Results` user interface. You can click 
         the :guilabel:`Messages` pill to see execution 
         statistics such as ``Execution Time``, ``Row Count``, 

>          error messages, and any print statements.

Reviewer 2: [minimalism]
```suggestion
         error messages, and print statements.
```

   .. step:: Run the converted query

      a. On the :guilabel:`Test Queries`
         pane, click :guilabel:`Run Converted Query` 
         button.

      #. Enter the connection details to your database.

      #. On the :guilabel:`Connection Details` form, click :guilabel:`Run`.

>          :guilabel:`Connection Details` form.

Reviewer 2: ```suggestion
         
```

         :guilabel:`Connection Details` form.

         The data from the destination database populates 

>          on the right-side of :guilabel:`Results`.

Reviewer 2: [suggestion]

Suggesting this for consistency on how this component is referred to through out the doc

```suggestion
         on the right-side of the :guilabel:`Results` pane.
```

         button.

      #. Enter the connection details to your source database.

      #. Click :guilabel:`Run` on the 
         :guilabel:`Connection Details` form.

         The data from the source database populates in the
         :guilabel:`Results` user interface. You can click 
         the :guilabel:`Messages` pill to see execution 
         statistics such as ``Execution Time``, ``Row Count``, 
         error messages, and print statements.
      
   .. step:: Run the converted query

      a. On the :guilabel:`Test Queries`
         pane, click :guilabel:`Run Converted Query` 
         button.

      #. Enter the connection details to your database.

      #. Click :guilabel:`Run` on the 
         :guilabel:`Connection Details` form.

         The data from the destination database populates 
         on the right-side of the :guilabel:`Results` user 
         interface.

      #. You can use the :guilabel:`Results` pane to compare the 
         source and destination data and types.

         .. note::

            You can click the :icon-lg:`Menu` and 
            :icon-lg:`Apps` buttons to switch between 
            the document and row view for the data. 

>    .. step:: (Optional) Edit the converted query

Author: Wow I had never seen this before, thanks for the reference.
